### PR TITLE
Redpanda 2.0.0 - Update for public_metrics and Redpanda Cloud

### PR DESCRIPTION
**Pull request by tmgstevens**
### What does this PR do?

Update the Redpanda integration to move from the legacy [metrics](https://docs.redpanda.com/docs/reference/internal-metrics-reference/) to the new [public_metrics](https://docs.redpanda.com/docs/reference/public-metrics-reference/) endpoint and to work with Redpanda Cloud.

### Motivation

Ability to monitor Redpanda Cloud with DataDog and also reduce the verbosity of metrics (improved use of histograms for example).

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

File: redpanda/manifest.json
Comment (yzhan289) [inline redpanda/manifest.json:23]: Do you need to also update the contact info in [CODEOWNERS](https://github.com/DataDog/integrations-extras/blob/d36f3b3683a68010ef33faa777cc6f3d896e207e/.github/CODEOWNERS#L131)?

**Review by yzhan289 — COMMENTED** [review]
Thanks for the update! Left 1 comment regarding CODEOWNERS. 

Pinging @rkruze since they own the integration. Once they approve, we can merge.

File: redpanda/manifest.json
Comment (tmgstevens) [reply redpanda/manifest.json:23]: Thanks @yzhan289 , updated that. Checking it needs to be a person rather than an organisation? I'm now seeing erros.

Comment (tmgstevens) [reply redpanda/manifest.json:23]: On https://github.com/DataDog/integrations-extras/pull/1864/files it showed CODEOWNERS errors, but the CI tests pass.

Comment (yzhan289) [reply redpanda/manifest.json:23]: I believe those errors are fine to ignore.

File: redpanda/assets/dashboards/overview.json
Comment (rkruze) [inline redpanda/assets/dashboards/overview.json:9]: @tmgstevens You might want to use a different source for the logo, like what was previously referenced which was in this github repo.

File: redpanda/assets/dashboards/overview.json
Comment (rkruze) [inline redpanda/assets/dashboards/overview.json:742]: @tmgstevens "Memory Utilization"

File: redpanda/assets/dashboards/overview.json
Comment (rkruze) [inline redpanda/assets/dashboards/overview.json:875]: "Bytes Sent"

Comment (tmgstevens) [reply redpanda/assets/dashboards/overview.json]: Fixed

File: redpanda/assets/dashboards/overview.json
Comment (tmgstevens) [reply redpanda/assets/dashboards/overview.json]: Updated

File: redpanda/assets/dashboards/overview.json
Comment (tmgstevens) [reply redpanda/assets/dashboards/overview.json:9]: This is our official logo source

**Review by rkruze — APPROVED** [review]

**Comment by rkruze** [comment]
@yzhan289 Approved

**Comment by yzhan289** [comment]
Thanks @rkruze and @tmgstevens!! 🙇

**Pull request merged**
